p2p/src/sync/peer_v1.rs

```diff
@@ -734,10 +733,28 @@ where
         // The code below will set it again if needed.
         self.peer_activity.set_expecting_blocks_since(None);
 
-        if self.incoming.requested_blocks.take(&block.get_id()).is_none() {
-            return Err(P2pError::ProtocolError(ProtocolError::UnexpectedMessage(
-                "block response".to_owned(),
-            )));
+        if self.incoming.requested_blocks.front().is_some_and(|id| id == &block.get_id()) {
+            self.incoming.requested_blocks.pop_front();
+        } else {
+            let idx = self.incoming.requested_blocks.iter().position(|id| id == &block.get_id());
+            // Note: we treat wrongly ordered blocks in the same way as unsolicited ones, i.e.
```

Comment:
Shouldn't we enforce this specifically in v2? Are we sure this is gonna be OK and not cause disconnections?

Reply:
Even in V1 we maintain the same order when sending blocks (`outgoing.blocks_queue` is already a `VecDeque`).
So no, there shouldn't be any disconnects because of this change.
It's just that this behaviour wasn't enforced, but now it is.